server/models/autoIncrementModel.js

```diff
@@ -0,0 +1,27 @@
+import mongoose from 'mongoose';
+
+const CounterSchema = new mongoose.Schema({
+  _id: { type: String, required: true },
+  seq: { type: Number, default: 0 },
+});
+const Counter = mongoose.model('Counter', CounterSchema);
+
+mongoose.set('useFindAndModify', false);
+
+async function increaseSeq(schema, key) {
+  const document = this;
+  await Counter.findByIdAndUpdate({ _id: schema }, { $inc: { seq: 1 } }, (error, doc) => {
+    if (error) {
```

Comment:
Use either async/Await or the callback function. I guess in this case you are using both.

Reply:
Done